### PR TITLE
PPTP-1076 : OrganisationDetails model refactor

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**
FE: https://github.com/hmrc/plastic-packaging-tax-registration-frontend/pull/282

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [x] Links to dependencies have been included (BE/FE work)
 - [x] User Acceptance Tests (UAT) were run locally and have passed.

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/models/RegistrationDetails.scala [hunk 1]
```diff
@@ -16,6 +16,18 @@
 
 package uk.gov.hmrc.plasticpackagingtaxregistration.models
 
-trait RegistrationDetails {
-  val registration: IncorporationRegistrationDetails
+import play.api.libs.json.{Json, OFormat}
+
+case class RegistrationDetails(
+  identifiersMatch: Boolean,
+  verificationStatus: String,
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/models/RegistrationDetails.scala:23]: If BV isn't called then there won't be a `verificationStatus` - something to bear in mind for later.

Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/models/RegistrationDetails.scala:23]: I thought if BV is not called the status is set to "UNCHALLENGED"?

Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/models/RegistrationDetails.scala]: We've made this optional.

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/base/data/RegistrationTestData.scala [hunk 2]
```diff
@@ -91,8 +92,8 @@ trait RegistrationTestData {
                                      lastName = "User",
                                      dateOfBirth = Some("1978-01-01"),
                                      nino = "567890123",
-                                     sautr = Some("123456789"),
```
Comment (tmc-ee) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/base/data/RegistrationTestData.scala:94]: Why was `sautr` removed here and do we need to do this for the details of the other entity types in this file? e.g. `sautr` and `ctutr` are used throughout.

Comment (RyanKellyHMRC) [reply test/uk/gov/hmrc/plasticpackagingtaxregistration/base/data/RegistrationTestData.scala:94]: Cheers Tom. I have no idea. I'll take a look...

**Review by tmc-ee — APPROVED** [review]

**Review by paulblakehmrc — APPROVED** [review]
LGTM but this needs to be followed up with a PR to prevent this running on each startup after the migration is done.

**Pull request merged**